### ISSUE TITLE
possible memory leak

### ISSUE THREAD

**Issue by deedarb** [vasyabigi/angular-slick#17]
Have you tested for memory leaks?
Official slick plugin suggests unslick() function to remove which is never called from your code.
Unfortunately code is written in coffeescript, no way to contribute :)
Directive should contain:
scope.$on("$destroy",function(){
      $(element).unslick();
});

**Comment by tilwinjoy** [comment]
I have tested for memory leak, and this does cause memoery leak. I have fixed this for now by by wrapping this in my own directive. Came here searching for existing issues before I make a PR, then noticed the source is  coffeescript as mentioned in OP.

Please add a destroy handler.

**Comment by tilwinjoy** [comment]
@grabbou 

> memory consumed by event handlers triggered on DOM elements is trivial 

Yes. But in angular this causes the detached elements to stay in memory, some of these elements holds references to scope(s), scope holds references to parent scopes, child scopes etc and they holds references to lots of data. Eventually those tiny handlers cause huge memory leak. This is a serious issue than just an improvement to code quality...
